www/scripts/productlist/ListOfProducts.js

```diff
@@ -213,20 +214,37 @@ function (declare, domClass, domConstruct, ItemFileWriteStore, topic,
         + name[0].toUpperCase()
         + '</span></div>';
 
-      if (!item.connected || !item.accessible) {
+      var dbStatus = item.databaseStatus;
+
+      var dbStatusMessages = {};
```

Comment:
I don't like this. Create a helper function in the util.js:
```js
    /**
     * Converts a Thrift API database status id to human readable string.
     *
     * @param {String|Number} dbStatus Thrift API DBStatus id.
     * @return Human readable review status string.
     */
    dbStatusFromCodeToString : function (dbStatus) {
      switch (parseInt(dbStatus)) {
        case DBStatus.OK:
          return 'Database is up to date.';
        case DBStatus.MISSING:
          return 'Database is missing.';
        case DBStatus.FAILED_TO_CONNECT:
          return 'Failed to connect to the database.';
        case DBStatus.SCHEMA_MISMATCH_OK:
          return 'Schema mismatch: migration is possible.';
        case DBStatus.SCHEMA_MISMATCH_NO:
          return 'Schema mismatch: migration not available.';
        case DBStatus.SCHEMA_MISSING:
          return 'Schema is missing.';
        case DBStatus.SCHEMA_INIT_ERROR:
          return 'Schema initialization error.';
        default:
          console.warn(
            'Non existing database status code: ',
            dbStatus);
          return 'N/A';
      }
    },
```
Use this function after it was added:
```js
var dbStatusMsg = util.dbStatusFromCodeToString(dbStatus);
```